Fix kubectl sort-by metadata issue.

aveshagarwal opened the pull request.

pmorie commented on pkg/kubectl/resource_printer.go:
```diff
@@ -424,6 +425,8 @@ var podSecurityPolicyColumns = []string{"NAME", "PRIV", "CAPS", "VOLUMEPLUGINS",
 func (h *HumanReadablePrinter) addDefaultHandlers() {
 	h.Handler(podColumns, printPod)
 	h.Handler(podColumns, printPodList)
+	h.Handler(podColumns, printPodv1)
```
This feels wrong to me

ncdc replied:
Ditto. We ideally shouldn't need version specific printers.

aveshagarwal commented:
@pmorie thanks. then how to print versioned objects? convert versioned objects again to internal types and print? But I am not sure how to do that in the code? any pointers?

pmorie commented:
@deads2k PTAL

ncdc commented:
Why do you need the json metadata to do the sorting?

pmorie commented:
I think that we need to apply sorting on versioned schema, but we should print the same stuff we print today.  To be fair, @aveshagarwal, I think this is a tricky issue to sort out :)

aveshagarwal commented:
As per https://github.com/kubernetes/kubernetes/blob/master/docs/user-guide/kubectl-overview.md:

```
kubectl [command] [TYPE] [NAME] --sort-by=<jsonpath_exp>
```

And so json metadata used by sort-by.

ncdc commented:
Sure, but is there code that looks at the json tags to do the sorting?

aveshagarwal commented:
yes in pkg/kubectl/sorting_printer.go.

aveshagarwal commented:
I am trying another approach where instead of printing a versioned object, I will try to print its corresponding internal object as sorter keeps original position of objects too.

aveshagarwal commented:
That seem to work (feeling why I did not think it before), will do more testing with origin and kube before submitting to kube upstream

aveshagarwal commented:
Updated PR, with some testing will send to upstream unless there is some more feedback. Probably thought too much about versioned objects that obscured this simple solution.

ncdc commented on pkg/kubectl/cmd/get.go:
```diff
@@ -262,10 +278,13 @@ func RunGet(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []string
 
 	for ix := range objs {
 		var mapping *meta.RESTMapping
+		var tmp runtime.Object
```
Call this something more informative that `tmp` that describes its purpose

Commit:
Fix kubectl sort-by metadata issue.

Internal types are not supposed to have json metadata (though in kubernetes
they do) as it is true with openshift types. That means sort-by must work
on versioned objects for sorting, otherwise it produces "error: metadata
is not found" error if it sorts internal types without json metadata.

This PR converts internal types objects to versioned objects and sort-by
sorts them correctly without medata error, and then it prints
corresponding internal objects in sorted order.

aveshagarwal commented:
I am closing this as I have sent it to kube upstream.
